File: AngularAspNetCoreChat/WebApi/ClientApp/src/app/models/message.model.ts
export class MessageModel {
  sender: string;
  reciever: string;
  text: string;
  date: Date;

  constructor(_sender: string, _reciever: string, _text: string, _date: Date) {
    this.sender = _sender;
    this.reciever = _reciever;
    this.text = _text;
    this.date = _date || new Date();
  }
}

File: NetCoreApi/WebApi/Models/MessageModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace WebApi.Models
{
    public class MessageModel
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public string Id { get; set; }
        public string Sender { get; set; }
        public string Reciever { get; set; }
        public string Text { get; set; }
        public DateTime Date { get; set; }
    }
}
